client/main.go

```diff
@@ -0,0 +1,1895 @@
+	responseJSON, err := json.Marshal(map[string]interface{}{
+		"can_download": canDownload,
+		"download_url": URL,
+	})
+	if err != nil {
+		http.Error(w, "Error converting to JSON: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(responseJSON)
+}
+
+// FetchGravatarImageHandler is a handler for the /profiles/fetch_gravatar_image endpoint.
+// It is used to fetch the Gravatar image for the user.
+func DownloadGravatarImageHandler(w http.ResponseWriter, r *http.Request) {
+	var data FetchGravatarData
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		http.Error(w, "Error parsing JSON: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	go DownloadGravatarImage(data)
+	w.WriteHeader(http.StatusOK)
+}
+
+// DownloadGravatarImage is a function for downloading the Gravatar image of the creator.
+// It first checks if Gravatar image is available locally. If not, it downloads it from the server.
+// It preferes to download the image from the server using the Avatar128 parameter,
+// but if it is not available, it tries to download it from Gravatar using gravatarHash.
+func DownloadGravatarImage(data FetchGravatarData) {
+	var url string
+	if data.Avatar128 != "" {
+		url = *Server + data.Avatar128
+	} else {
+		url = fmt.Sprintf("https://www.gravatar.com/avatar/%v?d=404", data.GravatarHash)
+	}
+
+	taskID := uuid.New().String()
+	AddTaskCh <- NewTask(data, data.AppID, taskID, "profiles/fetch_gravatar_image")
+
+	filename := fmt.Sprintf("%d.jpg", data.ID)
+	tempDir, err := GetSafeTempPath()
+	if err != nil {
+		TaskErrorCh <- &TaskError{AppID: data.AppID, TaskID: taskID, Error: err}
+		return
+	}
+
+	gravatarPath := filepath.Join(tempDir, gravatar_dirname, filename)
+	exists, _, _ := FileExists(gravatarPath)
+	if exists {
+		TaskFinishCh <- &TaskFinish{
+			AppID:   data.AppID,
+			TaskID:  taskID,
+			Message: "Found on disk",
+			Result:  map[string]string{"gravatar_path": gravatarPath},
+		}
+		return
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		TaskErrorCh <- &TaskError{AppID: data.AppID, TaskID: taskID, Error: err}
+		return
+	}
+
+	headers := getHeaders("", *SystemID, data.AddonVersion, data.PlatformVersion)
+	req.Header = headers
+	resp, err := ClientSmallThumbs.Do(req)
+	if err != nil {
+		TaskErrorCh <- &TaskError{AppID: data.AppID, TaskID: taskID, Error: err}
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		err = fmt.Errorf("error downloading gravatar image - %v: %v", resp.Status, url)
+		TaskErrorCh <- &TaskError{AppID: data.AppID, TaskID: taskID, Error: err}
+		return
+	}
+
+	// Open the file for writing
+	err = os.MkdirAll(filepath.Dir(gravatarPath), os.ModePerm)
+	if err != nil {
+		TaskErrorCh <- &TaskError{AppID: data.AppID, TaskID: taskID, Error: err}
+	}
```

Comment:
```suggestion
	if err != nil {
		TaskErrorCh <- &TaskError{AppID: data.AppID, TaskID: taskID, Error: err}
		return
	}
```
?

Reply:
Yeah, better to return directly, not to wait for os.Create() to return error because path does not exist. Gonna fix it.